demos/demo_non_Gauss_Likelihood.py

```diff
@@ -0,0 +1,63 @@
+# %%
```

Comment:
May be can add a brief description like:
"Demonstration showing a gradient based sampling using NUTs for two different Bayesian problems, both with non-Gaussian likelihoods."